python/delta_sharing/rest_client.py

```diff
@@ -273,8 +273,34 @@ def list_files_in_table(
                 add_files=[AddFile.from_json(json.loads(file)["file"]) for file in lines],
             )
 
+    @retry_with_exponential_backoff
     def list_table_changes(self, table: Table, cdfOptions: CdfOptions) -> ListTableChangesResponse:
-        raise HTTPError("API not implemented yet!")
+        query_str = f"/shares/{table.share}/schemas/{table.schema}/tables/{table.name}/changes?"
+
+        # Add cdf options.
+        # We do not validate the CDF options here since the server will perform validations anyways.
+        if cdfOptions.starting_version is not None:
```

Comment:
L282 to L290 is still assuming that one and only one starting parameter will be set, are you intended to do so?

I prefer we do not make such assumption? 

Also for starting_timestamp and ending_timestamp, we need to encode them for the server to be able to parse, something I found on web:

```
import urllib.parse
urllib.parse.quote(query)
```

Reply:
Removed the assumption.

I tried your suggestion about urllib, but the server does not like it and it returns an error.

Reply:
hmmm, I think we need to do sth like this, so that `&` is correctly added.

```
params_list = []
if cdfOptions.starting_version is not None:
  params_list.append(f"startingVersion={cdfOptions.starting_version}")
query_str = "&".join(params_list))
```

For the urllib, I think we need and only need to apply `urllib.parse.quote` on cdfOptions.starting_timestamp, so that it doesn't encode `=`, and it encodes the space ` `. 
This is due the the server behavior, verified in both Delta Sharing OSS server and our rpc parser: https://github.com/databricks/universe/pull/174415

And needs to add a corresponding test case to test the encoding behavior is correct in our client.

Reply:
Done... good catch on the params!